charts/render.py

```diff
@@ -114,25 +120,45 @@ def keep_doc(doc):
 
 
 def main():
-    (name, namespace, values, path) = sys.argv[1:]
+    parser = argparse.ArgumentParser()
+    parser.add_argument('name', help="Denotes the name of the chart")
+    parser.add_argument('namespace', help="Metalk8s namespace to deploy unto")
+    parser.add_argument('values', help="Our custom chart values")
+    parser.add_argument(
+        '--service-config',
+        action='append',
+        nargs=2,
+        dest="service_config",
+        help="Ex: --service-config service_name service_configmap_name"
+    )
+    parser.add_argument('path')
+    args, extra = parser.parse_known_args()
 
     template = subprocess.check_output([
         'helm', 'template',
-        '--name', name,
-        '--namespace', namespace,
-        '--values', values,
-        path,
+        '--name', args.name,
+        '--namespace', args.namespace,
+        '--values', args.values,
+        args.path,
     ])
 
     fixup = lambda doc: \
         fixup_metadata(
-            namespace=namespace,
+            namespace=args.namespace,
             doc=fixup_doc(
                 doc=doc
             )
         )
 
-    sys.stdout.write(BOILERPLATE.lstrip())
+    sys.stdout.write(
+        BOILERPLATE.format(
+            configlines='\n'.join(
+                "{} set {} = salt.metalk8s_service_configurator.get_service_conf('{}', '{}') {}".format(
+                    '{%-', p[0], args.namespace, p[1], '%}'
+                ) for p in args.service_config)
+        ).lstrip()
+    )
+    sys.stdout.write(SUB_BOILERPLATE)
```

Comment:
Not sure the split between `BOILERPLATE` and `SUB_BOILERPLATE` is needed or helpful. Actually, since we now template in some variable declarations, it's not only boilerplate anymore. Maybe just call them `[START|END]_TEMPLATE` or `[START|END]_BLOCK`?